tutor/templates/build/openedx/Dockerfile

```diff
@@ -144,38 +145,53 @@ RUN mkdir -p "$STATIC_ROOT_LMS"
 
 
 ###########################################################################################################################
-############## BUNDLES
+############## FRONTEND SOURCES
 ###########################################################################################################################
 
-FROM frontend-requirements as bundles
+# Intermediate stage to capture *only* frontend source files.
+# Frontend sources don't change very often, so this allows Docker to skip Webpack and Sass
+# compilation in most warm-cache situations.
+FROM frontend-requirements as frontend-sources
+
+# Copy in all edx-platform source dirs.
+COPY --link --from=edx-platform /lms lms
+COPY --link --from=edx-platform /cms cms
+COPY --link --from=edx-platform /common common
+COPY --link --from=edx-platform /openedx openedx
+COPY --link --from=edx-platform /xmodule xmodule
+
+# Now, delete everything that isn't relevant to the Webpack or Sass builds.
+# We expect that these patterns capture all the relevant static assets, although it is possible
+# that an upstream change could break that expectation. However, new frontends are generally
+# added to MFEs instead of edx-platform, so we don't expect much churn if any.
+RUN find lms cms common openedx xmodule -type f -a \! \
+	\( -path '*/static/*' \
+	-o -path '*/assets/*' \
+	-o -path '*/js/*' \
+	-o -name '*.underscore' \
+	\) -delete
```

Comment:
It's not perfect, but I think this list would be much more resilient to upstream changes than this list [would have been](https://github.com/kdmccormick/tutor/pull/34/files#r1316151049).